getnacelle/nacelle-js
# feat(gatsby-source-nacelle): update type defs

### WHY are these changes introduced?

Addresses [[ENG-4957](https://nacelle.atlassian.net/browse/ENG-4957)] 
Fixes [[ENG-4955](https://nacelle.atlassian.net/browse/ENG-4955)]

- Updates to the Storefront GraphQL need to be reflected in `gatsby-source-nacelle`.

### WHAT is this pull request doing?

- Updates Type Definitions to reflect new changes in Storefront GraphQL Schema.
- Updates required `@nacelle/storefront-sdk` version to be at least `1.1.2`.

## 

### How to Test

1. Run `npm run bootstrap`.
2. These updates will need to be tested in a Gatsby project. If you already have one available, run `npm link` from `packages/gatsby-source-nacelle` and then `npm link @nacelle/gatsby-source-nacelle` from your Gatsby project. If you do not have a Gatsby project, you can use the Gatsby starter in `starters/gatsby`.
3. Make sure to connect to a Nacelle Space, below are credentials to our Reference Store:
```
GATSBY_NACELLE_STOREFRONT_TOKEN=16f0d12a-7990-47dc-a5f0-4bd043e2b6e7
GATSBY_NACELLE_STOREFRONT_ENDPOINT=https://storefront.api.development.nacelle.com/graphql/v1/spaces/cky4xgh48965713nxjkv9s75uk
GATSBY_NACELLE_STOREFRONT_LOCALE="en-US"
```
4. Run `npm run clean` then `npm run develop`. 
5. Open the GraphiQL provided by Gatsby, typically at http://localhost:8000/___graphql and test queries for the updated fields.

Here's an example of retrieving the `featuredMedia` image source of a product collection from Package Free (related to issue [ENG-4955](https://nacelle.atlassian.net/browse/ENG-4955)):

<img width="800" alt="Screen Shot 2022-04-28 at 10 18 16 AM" src="https://user-images.githubusercontent.com/1411409/165810470-c8900e5f-272a-4fbf-8564-fe06b8ea62df.png">

### Testing checklist
